src/main/scala/com/databricks/spark/xml/util/TypeCast.scala

```diff
@@ -119,7 +119,15 @@ private[xml] object TypeCast {
       map(supportedXmlTimestampFormatters :+ _).getOrElse(supportedXmlTimestampFormatters)
     formatters.foreach { format =>
       try {
-        return Timestamp.from(ZonedDateTime.parse(value, format).toInstant)
+        // If format is not in supported and no timezone in format, use default Spark timezone
+        if (!supportedXmlTimestampFormatters.contains(format) && Option(format.getZone).isEmpty) {
```

Comment:
Rather than do this, just break this method into two checks - one loop over built-in formats, then the custom format with special TZ handling

Reply:
Do you mean we should use two methods? If I understand correctly,

```
    val formatters = options.timestampFormat.map(DateTimeFormatter.ofPattern).
      map(supportedXmlTimestampFormatters :+ _).getOrElse(supportedXmlTimestampFormatters)
    formatters.foreach { format =>
```

This block of code means that the current `parseXmlTimestamp()` method already loops over built-in formats + the custom format. It is just that if the specified format does not contains a timezone, the custom format does not work. So why break it if the original idea is to loop over all formats?

If you meant just break in the method, doesn't the if branch does it?

Reply:
It's all about the same, but why combine two types of input that need different processing and then re-differentiate in a loop? handle the loop, then handle the special case. I think it's more straightforward, maybe not.